### ISSUE TITLE
Adding Timeshift ( backup utility ) in our ISOs

### ISSUE THREAD

**Issue by Ste74** [manjaro/release-plan#170]
I have build and uploaded in unstable repo `timeshift-dev` . If someone don't know this program 
[here](https://github.com/teejee2008/timeshift/blob/master/README.md) what is and how work .
Since use rsync to work the size of snapshot is not big and also is possible use  a rotation for preserve the hd space. 
I think we can add an option on msm called Bachup or Manjaro-Backup to open the timeshift gui ( since not all know what is timeshift ) or since timeshift can be used via cli we can manage via msm the option .. 

thoughts ?

**Comment by codesardine** [comment]
Yes Please.

**Comment by alexblhr** [comment]
Yes ! Please :)

**Comment by Chrysostomus** [comment]
I would include it in some editions, but not all

**Comment by Ste74** [comment]
> I would include it in some editions, but not all

@Chrysostomus why ?

**Comment by Chrysostomus** [comment]
Because gut feeling :). I'm not yet convinced of its stability, even though it seems to have improved a lot.

For example in my gnome installation the launcher does not work, I need to launch it from cli

**Comment by Ste74** [comment]
This is strange, for me it start normally via desktop launcher .. you are in wayland ? if you launch via pkexec not work ?

**Comment by Chrysostomus** [comment]
It fails. Started to work now that I removed /etc/timeshift.json.

**Comment by Ste74** [comment]
Lol .. i warning in the forum ;) 
https://forum.manjaro.org/t/timeshift-v17-10-is-ready/32372

**Comment by Chrysostomus** [comment]
Tested timeshift now on new bspwm edition install. On the first run it creates configuration, then crashes. The gui will not start again. Deleting configuration lets you start gui again, but only for config generation. 

I'm not sure it is stable enough yet for all editions, but it would be really nice to have. It also has very low dependencies on gtk3 editions.

**Comment by Chrysostomus** [comment]
On another installation gui doesn't crash, but instead I get the error about unsupported subvolume layout (even though I have the required @ and @home subvolumes)

**Comment by Ste74** [comment]
At the moment Mate and Budgie ship with Timeshift

**Comment by Ste74** [comment]
@codesardine point me where you read this ( for no btrfs filesystem ) 😉

**Comment by codesardine** [comment]
@Ste74  I have not read it properly, apparently is only for btrfs

**Comment by Chrysostomus** [comment]
I recently got the issue narrowed down: for some reason, the btrfs setup generated by manjaro-architect is incompatible with timeshift. If I get that sorted out, then I would very much support adding timeshift to isos.

**Comment by Chrysostomus** [comment]
@Ste74 Do you have a working btrfs setup with timeshift? If so, could you post the output of 

     mount | grep btrfs
for troubleshooting purposes? It seems that using calamares style fstab is not enough. I wonder what is the issue...

**Comment by Ste74** [comment]
Unfortunally not use btrfs but i know an user with this set .. i can ask to send me the output if is ok for you

**Comment by Chrysostomus** [comment]
Would be great

**Comment by Ste74** [comment]
@Chrysostomus , there is the screenshot 
![photo_2018-02-12_18-51-55](https://user-images.githubusercontent.com/11218234/36111328-0c991c82-1026-11e8-84d1-038eb1d40f62.jpg)

**Comment by Chrysostomus** [comment]
The bug that caused issues with manjaro-architect has been found and is getting fixed with the next release of timeshift: 

https://github.com/teejee2008/timeshift/issues/9

So, after the next release I think we could include timeshift in profiles

**Comment by philmmanjaro** [comment]
I added this to **v18.0**. Would be great to know if it is still broken ...

**Comment by Chrysostomus** [comment]
Unfortunately it is, I tested it recently. Waiting for the next upstream update.

**Comment by Ste74** [comment]
@Chrysostomus you mean in btrfs case right ? btfrs is not our default files system so in my pov is a minor issue but i can wrong :wink: . For ext4 work better ..

**Comment by Chrysostomus** [comment]
Yeah, but we still support btrfs, right? And it is 50% of the timeshift functionality. I would still want to get it working. If I make manjaro-architect use only the ubuntu style @ and @home like calamares does, maybe it will do the trick... The setup is not as good, but m-a allows for custom subvolumes so it does not matter that much

**Comment by Ste74** [comment]
I see your report in timeshift repo and the answer of the developer.. it can be implement in next release but v18.2 not correct now this :| if you find a good workaround better for us and better for the flexibility of m-a :)

**Comment by Chrysostomus** [comment]
Having just @ and @home is not enough. I'll try to determine the difference....

**Comment by Chrysostomus** [comment]
Okay, I now managed to get it working with m-a based installation. So, it can be done, just needs some tweaking in m-a. Ready to be included I would say.

**Comment by philmmanjaro** [comment]
@Chrysostomus: please add the needed modifications to **m-a** as needed. Do we have to adjust some in our **manjaro-tools** or only update the **iso-profiles**?

**Comment by Chrysostomus** [comment]
Manjaro-architect in iso-profiles is self updating, so we only need to add timeshift. No need for other changes.

I'll try to make the changes in time for the next release, but this is the second busiest week of the year, so I cannot guarantee the success. But even if we release the isos first, manjaro-architect will be updated on use anyway.

**Comment by philmmanjaro** [comment]
@Chrysostomus: there is no rush as we are scheduled for May.

**Comment by Chrysostomus** [comment]
Okay, manjaro-architect now supports timeshift in btrfs mode. I'll update the package now. 

Some things I noticed about timeshift:
- It seems to enable quotas in btrfs subvolumes. Afaik, this is an unstable btrfs feature. This is probably approriate to discuss in timeshift git repo
- The package has hidden dependencies. It uses polkit and udisks2 internally, but does not depend on them on the package level. Polkit is a bit complicated because polkit agents do not provide any single package, but we should at least add the udisks2 dependency. This is relevant because we have editions without polkit or udisks2

**Comment by Chrysostomus** [comment]
Update manjaro-architect has now been pushed to unstable and testing

**Comment by Chrysostomus** [comment]
The issues in timeshift should now be solved, so I can revert m-a to previous behaviour.

**Comment by Ste74** [comment]
I can't update timeshift for problem during compiling .. seem vala 0.40.2 introduce a regression already fixed but not released in **0.40.3** .. @philmmanjaro waiting for **0.40.3** in arch repo or reintroduce temporary **0.38 series** for our building ?

**Comment by Ste74** [comment]
Tried to build with `vala 0.38.8` but again build fails but now with less error so the problem not is only new vala.. hope @teejee2008 look into it
